app/models/generic_object_definition.rb

```diff
@@ -192,6 +237,91 @@ def validate_property_attributes
     end
   end
 
+  def validate_property_attribute_constraints
+    properties[:attribute_constraints].each do |attr_name, constraints|
```

Comment:
Is it possible for `properties[:attribute_constraints]` to be `nil` here for existing `generic_object_definitions` unless we're also doing a data migration?